File: Algorithms/Implementation/Cats and a Mouse/Cat.java
import java.util.Scanner;

class Cat {
	static String catAndMouse(int x, int y, int z) {
		int catA = Math.abs(x - z);
		int catB = Math.abs(y - z);

		if (catA == catB) {
			return "Mouse C";
		} else {
			return (catA < catB ? "Cat A" : "Cat B");
		}
	}

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int q = in.nextInt();

		for (int i = 0; i < q; i++) {
			in.skip("\n");
			int x = in.nextInt();
			int y = in.nextInt();
			int z = in.nextInt();

			System.out.println(catAndMouse(x, y, z));
		}
	}
}

File: Algorithms/Implementation/Cats and a Mouse/cat.cpp
#include <iostream>
#include <cstdlib>

std::string catAndMouse(int x, int y, int z) {
	int catA = abs(x - z);
	int catB = abs(y - z);

	if (catA == catB) {
		return "Mouse C";
	} else {
		return (catA < catB ? "Cat A" : "Cat B");
	}
}

int main()
{
	int q = 0;
	std::cin >> q;

	int x = 0, y = 0, z = 0;
	for (int i = 0; i < q; i++) {
		std::cin >> x >> y >> z;
		std::cout << catAndMouse(x, y , z) << std::endl;
	}
}